core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -200,21 +203,21 @@ protected void add(DataFile file) {
    * Add a delete file to the new snapshot.
    */
   protected void add(DeleteFile file) {
-    setWriteSpec(file);
-    addedFilesSummary.addedFile(writeSpec(), file);
+    Preconditions.checkNotNull(file, "Invalid delete file: null");
+    PartitionSpec fileSpec = ops.current().spec(file.specId());
+    List<DeleteFile> deleteFiles = newDeleteFiles.computeIfAbsent(file.specId(), specId -> Lists.newArrayList());
```

Comment:
Not a big deal, but for me this would be easier to understand if it was deleteFilesForSpec

Reply:
Sure, I'll update that. You refer to the map name, right?

Reply:
Updated the map name.